index.bs

```diff
@@ -87,8 +87,40 @@ never be allowed using this API, rather than leaving it entirely up to underlyin
 systems.
 
 A <dfn lt="file|file entry">file entry</dfn> additionally consists of
-<dfn for="file entry">binary data</dfn> (a [=byte sequence=]) and a
-<dfn for="file entry">modification timestamp</dfn> (a number representing the number of milliseconds since the <a spec=FileAPI>Unix Epoch</a>).
+<dfn for="file entry">binary data</dfn> (a [=byte sequence=]), a
+<dfn for="file entry">modification timestamp</dfn> (a number representing the number of milliseconds since the <a spec=FileAPI>Unix Epoch</a>)
+and a <dfn for="file entry">lock</dfn>. A lock is a string that may exclusively be "`open`", "`taken-exclusive`", and "`taken-shared`".
+
+<div algorithm>
+To <dfn for="file entry/lock">take</dfn> a [=file entry/lock=] with a |value| of "`exclusive`" or "`shared`" on a given [=file entry=] |file|,
+run the following steps:
+
+1. Let |lock| be the |file|'s [=file entry/lock=].
+1. If |value| is "`exclusive`":
+   1. If |lock| is "`open`":
+     1. Set lock to "`taken-exclusive`".
+     1. Return `true`.
+1. If |value| is "`shared`":
+   1. If |lock| is "`open`":
+     1. Set lock to "`taken-shared`".
+     1. Return `true`.
+   1. Else if |lock| is "`taken-shared`":
+     1. Return `true`.
+1. Return `false`.
+
+</div>
+
+<div algorithm>
+To <dfn for="file entry/lock">release</dfn> a [=file entry/lock=] on a given [=file entry=] |file|,
+run the following steps:
+
+1. Let |lock| be the |file|'s associated [=file entry/lock=].
+1. Set |lock| to "`open`".
```

Comment:
This says that if 10 WritableFileStreams are created with shared locks, and one of the WritableFileStreams goes away, it will cause the lock to be dropped.

Though I'll also note that it's unclear *how* a WritableFileStream's being destroyed would lead to a 'release' occurring -- unlike SyncAccessHandles, there's no 'close', so it would be dependent on garbage collection, and we generally want GC to not be visible to applications.  Alternatively, this may mean that once a WritableFileStream is created for a file, you can never get a SyncAccessHandle to that file (unless the browser is restarted, or perhaps after all instances of that origin have gone away).
In any case, this is confusing and needs to be much more clearly specified.

Reply:
Also odd with WritableFileStreams: they support (only) write(), seek() and truncate() (and they keep the "writes to a backup of the file).   No way to read the data (except with an SyncAccessHandle).  And no way to read an OPFS file except on a Worker (since the non-sync AccessHandle seems to have disappeared).